### PR TITLE
CircleCI - Add trigger workflow command

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
- Added the command ***circleci-trigger-workflow***.
- Added the *branch* argument in the ***circleci-workflow-last-runs command*** command.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.yml [hunk 1]
```diff
@@ -274,6 +274,12 @@ script:
       name: project
       required: false
       secret: false
+    - default: false
+      description: Name of github branch.
```
Comment (yaakovi) [inline Packs/CircleCI/Integrations/CircleCI/CircleCI.yml:278]:
```suggestion
      description: Name of the branch.
```

File: Packs/CircleCI/ReleaseNotes/1_0_2.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### CircleCI
+- Added the command ***circleci-trigger-workflow***.
+- Added the *branch* argument in the ***circleci-workflow-last-runs command*** command.
```
Comment (yaakovi) [inline Packs/CircleCI/ReleaseNotes/1_0_2.md:5]:
```suggestion
- Added the ***circleci-trigger-workflow*** command.
- Added the *branch* argument in the ***circleci-workflow-last-runs command*** command.
```

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py [hunk 3]
```diff
@@ -51,6 +59,19 @@ def get_workflow_jobs(self, workflow_id: str, page_token: Optional[str] = None):
             params={'page-token': page_token} if page_token else None
         )
 
+    def trigger_workflow(self, vc_type: str, organization: str, project: str, parameters: str):
+        headers = self._headers
+        headers['Circle-Token'] = demisto.params().get('api_key')
```
Comment (yaakovi) [inline Packs/CircleCI/Integrations/CircleCI/CircleCI.py:64]: why do you need to add it here? the headers already include the API key under the `Authentication` header.
in any case, you shouldn't use `demisto.params` here, the value is already passed to the client.

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py [hunk 4]
```diff
@@ -36,11 +36,19 @@ def get_workflows_list(self, vc_type: str, organization: str, project: str, page
 
     def get_last_workflow_runs(self, vc_type: str, organization: str, project: str, workflow_name: str,
                                page_token: Optional[str] = None):
+        branch: str = demisto.args().get('branch', '')
```
Comment (yaakovi) [inline Packs/CircleCI/Integrations/CircleCI/CircleCI.py:39]: it should be passed from the `circleci_workflow_last_runs_command` function

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py [hunk 5]
```diff
@@ -245,6 +266,26 @@ def circleci_workflow_last_runs_command(client: Client, args: Dict[str, Any]) ->
...
 
 
+def circleci_trigger_workflow_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    vc_type, organization, project, _ = get_common_arguments(client, args)
+    parameters_json: str = args.get('parameters', '')
+
+    try:
+        parameters = json.loads(parameters_json)
+    except ValueError:
+        raise DemistoException("Failed to parse parameters argument")
```
Comment (yaakovi) [inline Packs/CircleCI/Integrations/CircleCI/CircleCI.py:276]:
```suggestion
        raise DemistoException("Failed to parse the 'parameters' argument.")
```

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py (hunk 4 above)
Comment (adi88d) [reply Packs/CircleCI/Integrations/CircleCI/CircleCI.py]: done

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py (hunk 3 above)
Comment (adi88d) [reply Packs/CircleCI/Integrations/CircleCI/CircleCI.py]: The headers include the API key as basic authentication, when I tried to call the command with this headers it failed with this error:
`Failed to execute circleci-trigger-workflow command.
Error:
Error in API call [403] - Forbidden
{"message": "Permission denied"}`

for calling the trigger a workflow request we need to use [api_key_header](https://circleci.com/docs/api/v2/#section/Authentication/api_key_header).

I changed the code to save the API key when initialize the client and use it inside `trigger_workflow` method.

For more information:
https://support.circleci.com/hc/en-us/articles/360050351292-How-to-trigger-a-workflow-via-CircleCI-API-v2

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py [hunk 6]
```diff
@@ -245,6 +265,26 @@ def circleci_workflow_last_runs_command(client: Client, args: Dict[str, Any]) ->
...
+        raise DemistoException("Failed to parse the 'parameters' argument.")
+
+    response_json = client.trigger_workflow(vc_type, organization, project, parameters)
+    response = json.loads(response_json)
+
+    return CommandResults(
+        outputs_prefix='CircleCI.WorkflowTrigger',
+        outputs_key_field='id',
+        readable_output=f"CircleCI Workflow created successfully, ID={response.get('number')}",
+        outputs=response
```
Comment (yaakovi) [inline Packs/CircleCI/Integrations/CircleCI/CircleCI.py:284]:
```suggestion
        outputs=response,
```

File: Packs/CircleCI/Integrations/CircleCI/CircleCI.py [hunk 7]
```diff
@@ -51,6 +59,17 @@ def get_workflow_jobs(self, workflow_id: str, page_token: Optional[str] = None):
...
 
+    def trigger_workflow(self, vc_type: str, organization: str, project: str, parameters: str):
+        url_suffix = f'project/{vc_type}/{organization}/{project}/pipeline'
+
+        return self._http_request(
+            method='POST',
+            url_suffix=url_suffix,
+            json_data=parameters,
+            resp_type='text',
+            headers={'Circle-Token': self.api_key}
```
[the remark this answers is not in the document]
Comment (adi88d) [reply Packs/CircleCI/Integrations/CircleCI/CircleCI.py:70]: for this command no

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

**Pull request merged**